### PR TITLE
Add get_supermajority_slot() function

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
There's currently no way to calculate the latest slot on a fork that has reached supermajority lockout, which is needed for figuring out when to squash on ledger replay.

#### Summary of Changes
Introduce a function in the bank to calculate supermajority lockout

Fixes #

### CODE REVIEW

**Comment by aeyakovenko** [comment]
A slot with max lockout means that any ansestors have a max lockout as well.

**Comment by garious** [comment]
Cool, but first, can you create a PR that moves `vote_states` to a new `src/vote_utils.rs` module so that this PR doesn't need to add a bunch of vote-aware code to the solana_runtime crate?

**Comment by rob-solana** [comment]
> The “active” set from which to calculate the supermajority should be the same as for leader schedule. So calculated at an epoch.

same as saying that stakes not included in the leader schedule don't count when calculating supermajority?

**Comment by garious** [comment]
@carllin, can you rebase?

**Comment by carllin** [comment]
Rebased :)

**Comment by garious** [comment]
@aeyakovenko, any remaining concerns with this one?

File: src/staking_utils.rs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+{
+    let slot_height = current_slot_height.saturating_sub(bank.stakers_slot_offset());
+
+    let parents = bank.parents();
+    let mut banks = vec![bank];
+    banks.extend(parents.iter().map(|x| x.as_ref()));
+
+    let bank = banks
+        .iter()
+        .find(|bank| bank.slot_height() <= slot_height)
```
Comment (rob-solana) [inline src/staking_utils.rs]: prefer you use bank.id() over bank.slot_height().  slot_height() is calculated from tick_height, which is often incorrect for a fresh bank.  also, we only want frozen banks, but I think we're getting that by accident?

I'm not sure I'm comfortable with an approach that relies on other parts of the system not squash()ing.  What would make me more comfortable is to have stake should be calculated as we freeze(), and cached somewhere outside the Bank.

**Pull request merged**